--- origsrc/notification-daemon-0.3.7/src/daemon/engines.c	2007-02-28 01:18:48.000000000 -0600
+++ src/notification-daemon-0.3.7/src/daemon/engines.c	2007-06-01 20:31:17.625000000 -0500
@@ -40,12 +40,9 @@
 load_theme_engine(const char *name)
 {
 	ThemeEngine *engine;
-	char *filename;
 	char *path;
 
-	filename = g_strdup_printf("lib%s.so", name);
-	path = g_build_filename(ENGINES_DIR, filename, NULL);
-	g_free(filename);
+	path = g_module_build_path(ENGINES_DIR, name);
 
 	engine = g_new0(ThemeEngine, 1);
 	engine->ref_count = 1;
@@ -96,7 +93,7 @@
 
 error:
 	if (engine->module != NULL && !g_module_close(engine->module))
-		g_warning("%s: %s", filename, g_module_error());
+		g_warning("%s: %s", name, g_module_error());
 
 	g_free(engine);
 	return NULL;
--- origsrc/notification-daemon-0.3.7/src/themes/bubble/Makefile.am	2006-10-08 16:04:02.000000000 -0500
+++ src/notification-daemon-0.3.7/src/themes/bubble/Makefile.am	2007-06-01 20:31:17.625000000 -0500
@@ -9,7 +9,8 @@
 	eggnotificationbubblewidget.c \
 	theme.c
 
-libbubble_la_LDFLAGS = -module -avoid-version
+libbubble_la_LDFLAGS = -module -avoid-version -no-undefined
+libbubble_la_LIBADD = $(NOTIFICATION_DAEMON_LIBS)
 
 INCLUDES = \
 	$(NOTIFICATION_DAEMON_CFLAGS)
--- origsrc/notification-daemon-0.3.7/src/themes/standard/Makefile.am	2006-10-08 16:04:02.000000000 -0500
+++ src/notification-daemon-0.3.7/src/themes/standard/Makefile.am	2007-06-01 20:31:17.625000000 -0500
@@ -3,7 +3,7 @@
 engine_LTLIBRARIES = libstandard.la
 
 libstandard_la_SOURCES = theme.c
-libstandard_la_LDFLAGS = -module -avoid-version
+libstandard_la_LDFLAGS = -module -avoid-version -no-undefined
 libstandard_la_LIBADD  = $(NOTIFICATION_DAEMON_LIBS)
 
 INCLUDES = $(NOTIFICATION_DAEMON_CFLAGS)
